### PR TITLE
WIP Added buttons to user profile page for GH/OSO link/unlink

### PR DESCRIPTION
**Pull request by sbryzak**
### Work in progress

This PR adds buttons to the user profile page, allowing the connected GitHub and Openshift accounts to be (re)linked/unlinked.  See this issue for more details:

[](https://github.com/fabric8-services/fabric8-auth/issues/151)

Some styling is still required to position buttons correctly.

These changes depend on the following PR in the ngx-login-client module:

[](https://github.com/fabric8-ui/ngx-login-client/pull/94)

Signed-off-by: Shane Bryzak <sbryzak@redhat.com>

### CODE REVIEW

File: src/app/profile/overview/connected-accounts/connected-accounts.component.html [hunk 1]
```diff
@@ -41,10 +41,20 @@ <h2 class="card-pf-title">
                         <span>
                           Connected
                         </span>
+                        <span>
```
Comment (mindreeper2420) [inline src/app/profile/overview/connected-accounts/connected-accounts.component.html]: you can remove these `<span>` tags

File: src/app/profile/overview/connected-accounts/connected-accounts.component.html [hunk 2]
```diff
@@ -41,10 +41,20 @@ <h2 class="card-pf-title">
...
+                          <button class="btn btn-lg btn-default" id="gh_unlink"
+                          (click)="disconnectGitHub()">Disconnect</button>
+                        </span>
                       </ng-template>
                       <ng-template #ghUnlinkedUsername>
                         <span class="account-disconnected">
-                          Not Connected
+                          Disconnected
+                        </span>
+                        <span>
```
Comment (mindreeper2420) [inline src/app/profile/overview/connected-accounts/connected-accounts.component.html]: can also be removed

File: src/app/profile/overview/connected-accounts/connected-accounts.component.html [hunk 3]
```diff
@@ -77,11 +87,21 @@ <h2 class="card-pf-title">
                         <span>
                           {{contextUserName}}
                         </span>
+                        <span>
```
Comment (mindreeper2420) [inline src/app/profile/overview/connected-accounts/connected-accounts.component.html]: remove as well

File: src/app/profile/overview/connected-accounts/connected-accounts.component.html [hunk 4]
```diff
@@ -77,11 +87,21 @@ <h2 class="card-pf-title">
...
+                          (click)="connectOpenShift()">Reconnect</button>
+                          <button class="btn btn-lg btn-default" id="oso_unlink"
+                          (click)="disconnectOpenShift()">Disconnect</button>
+                        </span>
                       </ng-template>
                       <ng-template #osoUnlinkedUsername>
                         <span class="account-disconnected">
                           Not Connected
                         </span>
+                        <span>
```
Comment (mindreeper2420) [inline src/app/profile/overview/connected-accounts/connected-accounts.component.html]: last one to remove

File: src/app/profile/overview/connected-accounts/connected-accounts.component.html [hunk 5]
```diff
@@ -41,10 +41,20 @@ <h2 class="card-pf-title">
                         <span>
```
Comment (mindreeper2420) [inline src/app/profile/overview/connected-accounts/connected-accounts.component.html]: add class `margin-right-15` here to add a nice space between this text and the button group.

File: src/app/profile/overview/connected-accounts/connected-accounts.component.html [hunk 6]
```diff
@@ -41,10 +41,20 @@ <h2 class="card-pf-title">
...
                         </span>
+                        <span>
+                          <button class="btn btn-lg btn-default" id="gh_relink"
+                          (click)="connectGitHub()">Reconnect</button>
+                          <button class="btn btn-lg btn-default" id="gh_unlink"
+                          (click)="disconnectGitHub()">Disconnect</button>
+                        </span>
                       </ng-template>
                       <ng-template #ghUnlinkedUsername>
                         <span class="account-disconnected">
```
Comment (mindreeper2420) [inline src/app/profile/overview/connected-accounts/connected-accounts.component.html]: same here - add class `margin-right-15` to apply a space between this text and the button

File: src/app/profile/overview/connected-accounts/connected-accounts.component.html [hunk 7]
```diff
@@ -77,11 +87,21 @@ <h2 class="card-pf-title">
                         <span>
```
Comment (mindreeper2420) [inline src/app/profile/overview/connected-accounts/connected-accounts.component.html]: add `class="margin-right-15"`

File: src/app/profile/overview/connected-accounts/connected-accounts.component.html [hunk 8]
```diff
@@ -77,11 +87,21 @@ <h2 class="card-pf-title">
...
                         </span>
+                        <span>
+                          <button class="btn btn-lg btn-default" id="oso_relink"
+                          (click)="connectOpenShift()">Reconnect</button>
+                          <button class="btn btn-lg btn-default" id="oso_unlink"
+                          (click)="disconnectOpenShift()">Disconnect</button>
+                        </span>
                       </ng-template>
                       <ng-template #osoUnlinkedUsername>
                         <span class="account-disconnected">
```
Comment (mindreeper2420) [inline src/app/profile/overview/connected-accounts/connected-accounts.component.html]: add `class="margin-right-15"`

**Comment by sbryzak** [comment]
I've updated the button styles, and removed the extraneous <span> elements.  I've also added the margin-right-15 class to the "reconnect" buttons in order to place a gap between the two buttons.  I'm still working on updating the UI upon a successful disconnect of either accounts, however I've been stuck for quite a few hours on getting this to happen in the authentication service.

**Comment by joshuawilson** [comment]
please rebase

**Comment by joshuawilson** [comment]
@sbryzak what is the status of this?

**Comment by alexeykazakov** [comment]
We can close this. @dgutride is working on this.

**Pull request closed without merging**